src/Main.cpp

```diff
@@ -120,6 +120,7 @@ class MessageDispatcherTask : public AbstractTask<10 * configMINIMAL_STACK_SIZE>
     }
 };
 
+template <typename SerializerType>
```

Comment:
Is there a reason for a template instead of injecting a reference of the common interface? By this I mean a `IHiveMindHostSerializer`. It's already created in the comm monitoring, so you could pass this pointer instead of the stream

Reply:
Not really sure how to do this since the MessageSenderTask is injected in the comm monitoring. I would need to set it with a setter? If so, a reference would not work, it would have to be a pointer...

Reply:
Yes, as of now we are using a pointer for the stream in the message sender and then create a serializer, so instead we could use a pointer to the interface of serializer.